Wikipedia/Code/WMFCVLSection.m

```diff
@@ -143,16 +143,25 @@ - (CGFloat)setSize:(CGSize)size forItemAtIndex:(NSInteger)index invalidationCont
 
 - (void)offsetHeadersStartingAtIndex:(NSInteger)headerIndex distance:(CGFloat)deltaY invalidationContext:(WMFCVLInvalidationContext *)invalidationContext {
     NSArray *invalidatedHeaderIndexPaths = [self offsetAttributesInArray:_headers startingAtIndex:headerIndex distance:deltaY];
+    if (invalidatedHeaderIndexPaths.count == 0) {
```

Comment:
It's an optimization I added when I noticed the zero length case happening when debugging another issue which turned out to be unrelated. However, with collection view layouts, I figure you can never be too cautious.